numpy/core/tests/test_datetime.py

```diff
@@ -903,33 +909,29 @@ def test_datetime_add(self):
             assert_equal(tda + tdb, tdc)
             assert_equal((tda + tdb).dtype, np.dtype('m8[h]'))
             # m8 + bool
-            assert_equal(tdb + True, tdb + 1)
+            assert_equal(tdb + True, tdb + np.timedelta64(1, 'h'))
             assert_equal((tdb + True).dtype, np.dtype('m8[h]'))
-            # m8 + int
-            assert_equal(tdb + 3*24, tdc)
-            assert_equal((tdb + 3*24).dtype, np.dtype('m8[h]'))
+            # m8 + int prohibited
+            assert_raises(TypeError, np.add, tdb, 3*24)
             # bool + m8
             assert_equal(False + tdb, tdb)
             assert_equal((False + tdb).dtype, np.dtype('m8[h]'))
-            # int + m8
-            assert_equal(3*24 + tdb, tdc)
-            assert_equal((3*24 + tdb).dtype, np.dtype('m8[h]'))
+            # int + m8 prohibited
+            assert_raises(TypeError, np.add, 3*24 , tdb)
             # M8 + bool
-            assert_equal(dta + True, dta + 1)
+            assert_equal(dta + True, dta + np.timedelta64(1, 'D'))
```

Comment:
I'm a little suspicious that we'd want to prohibit integer operations with m8 because of the integer sloppiness, but allow operations with bool, but I'm sure I'll get feedback on that.

Reply:
I agree, it makes sense to me to keep them consistent with each other.

Reply:
ok, I've banned the bool casting / interaction with timedelta64 as well